cmd/validator/core/validator.go

```diff
@@ -241,6 +345,34 @@ func getData(url, token string) ([]byte, error) {
 	return data, ioErr
 }
 
+func resultResourceError(err error, resource string) Result {
+	return Result{
+		Equal:   false,
+		Error:   err,
+		Retries: 0,
+		Details: &ErrorDetails{
+			Resource: resource,
+		},
+	}
+}
+func resultBoxError(err error, appid string, boxname string) Result {
+	return resultBoxSkip(err, appid, boxname, NotSkipped)
+}
+
+func resultBoxSkip(err error, appid string, boxname string, skip Skip) Result {
+	return Result{
+		Equal:      false,
+		Error:      err,
+		SkipReason: skip,
+		Retries:    0,
+		Details: &ErrorDetails{
+			Resource: "box",
+			Appid:    appid,
+			Boxname:  boxname,
+		},
+	}
```

Comment:
nit: this isn't as much of an improvement as I thought it might be. Take it or leave it
```suggestion
	result := resultResourceError(err, "box")
	result.SkipReason = skip
	result.Details.Appid = appid,
	result.Details.Boxname = boxname,
	return result
```